src/main/java/chess/ChessService.java

```diff
@@ -98,22 +104,42 @@ private ChessBoard loadChessBoard(long roomId) {
         return new ChessBoard(() -> board);
     }
 
-    public Status status(String roomName) {
-        Room room = roomDao.findByName(roomName)
+    public Status status(long roomId) {
+        Room room = roomDao.findById(roomId)
             .orElseThrow(() -> new NoSuchElementException(NO_ROOM_MESSAGE));
         ChessBoard chessBoard = loadChessBoard(room.getId());
         ChessGame chessGame = ChessGame.of(chessBoard, room.getTurn());
 
         return chessGame.status();
     }
 
-    public boolean createRoom(String name) {
+    public Room createRoom(String name, String password) {
         Optional<Room> room = roomDao.findByName(name);
-        if (room.isEmpty()) {
-            Room newRoom = new Room(name);
-            roomDao.save(newRoom);
-            return true;
+        if (room.isPresent()) {
+            throw new IllegalArgumentException(DUPLICATED_ROOM_NAME_MESSAGE);
         }
-        return false;
+        Room newRoom = new Room(name, password);
+        roomDao.save(newRoom);
+        return roomDao.findByNameAndPassword(name, password).get();
+    }
+
+    public List<RoomDto> findAllRooms() {
+        return roomDao.findAll().stream()
+            .map(room -> new RoomDto(room.getId(), room.getTurn(), room.getName()))
```

Comment:
Dto와 Room 도메인을 분리하기 위해서 게터를 사용하여 생성해주시는군요! 

도메인은 dto를 알면 안되지만 dto는 도메인을 알아도 된다고 생각해요. 

아래와 같이 수정하여 보다 깔끔하게 구성할거 같습니다.

```suggestion
            .map(RoomDto::from)
```

```java
// RoomDto.java

    public static RoomDto from(Room room) {
        return new RoomDto(room.getId(), room.getTurn(), room.getName());
    }
```

Reply:
BoardDto에서는 static factory method를 활용하고 RoomDto에서는 getter를 client에서 사용하여 생성하고 있었군요!! RoomDto 생성을 내부로 감추도록 변경하였습니다.